projects/plugins/jetpack/extensions/shared/register-jetpack-block.js

```diff
@@ -59,6 +62,24 @@ export default function registerJetpackBlock( name, settings, childBlocks = [],
 	return result;
 }
 
+// Localize properties passed as parameter when applicable.
+// Note: the documentation states that `registerBlockType` wraps localized properties in `_x`
+// when it is passed the metadata object as the first param. I haven't found this to be true
+// with the latest version of `@wordpress/blocks` (12.18.0) at the time of writing this.
+// https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#javascript
+const localizeBlockProperties = ( metadata, props ) => {
+	const result = { ...props };
+
+	for ( const prop in props ) {
+		if ( LOCALIZED_BLOCK_PROPERTIES.includes( prop ) ) {
+			// eslint-disable-next-line
+			result[ prop ] = _x( props[ prop ], null, metadata.textdomain || DEFAULT_TEXTDOMAIN );
```

Comment:
(most of this is already known to the people involved via Slack discussions, but to close the loop I'll repeat it here)

I think this only accidentally works if your po files still have the translations from https://github.com/Automattic/jetpack/blob/427e370fe89c3ed1944a1ba93e97be94f00fd034/projects/plugins/jetpack/extensions/blocks/business-hours/index.js#L76-L77 that were removed in #32698.

WordPress intends for the translation to happen on the PHP side during block registration ([here](https://github.com/WordPress/WordPress/blob/f6307ff5fcfde0081cf94194c56defb8370b8c94/wp-includes/blocks.php#L443-L450)), then the already-translated texts are sent to JS via a generated call to `wp.blocks.unstable__bootstrapServerSideBlockDefinitions` ([here](https://github.com/WordPress/wordpress-develop/blob/9db15f2863e3625451d929767253bb1bb6c97388/src/wp-admin/site-editor.php#L129-L132), [here](https://github.com/WordPress/wordpress-develop/blob/9db15f2863e3625451d929767253bb1bb6c97388/src/wp-admin/edit-form-blocks.php#L108-L111), [here](https://github.com/WordPress/wordpress-develop/blob/9db15f2863e3625451d929767253bb1bb6c97388/src/wp-admin/widgets-form-blocks.php#L49-L52), or [here](https://github.com/WordPress/wordpress-develop/blob/9db15f2863e3625451d929767253bb1bb6c97388/src/wp-includes/class-wp-customize-widgets.php#L864-L867)).

That PHP code path is only reached when using `register_block_type_from_metadata()`, or passing a path to the block.json to `register_block_type()`. We instead [pass a WP_Block_Type object](https://github.com/Automattic/jetpack/blob/57c470f7a4ad3e87fb76d4f75df86d442c414ee2/projects/packages/blocks/src/class-blocks.php#L53-L65). So we'd either need to change our code to pass the block.json path, or duplicate the relevant logic (plus any other needed logic for any other core block.json registeration features) when creating our `WP_Block_Type`.